### DISCUSSION TITLE
Since 1.99.0 scanning for new content i my external library is no longer working

### DISCUSSION THREAD

**Discussion by kingvuke** [immich-app/immich#8286] [Q&A]
Hello everyone,
Since updating to 1.99, scanning for my external library is no longer working for me. I have upgraded from 1.97.X or 1.98.X to 1.99. I can't remember the exact version before that.

I got this in the Logs
`[Nest] 187  - 03/26/2024, 9:58:37 AM     LOG [ImmichMicroservice] Immich Microservices is listening on http://[::1]:3002 [v1.99.0] [PRODUCTION] 
[Nest] 187  - 03/26/2024, 10:16:55 AM   ERROR [LibraryService] Skipping invalid import path: /photos_library. Reason: Cannot use media upload folder for external libraries
[Nest] 160  - 03/26/2024, 10:18:18 AM     LOG [CommunicationRepository] Websocket Connect:    6YxErQmfn4YVmOY9AAAF
[Nest] 187  - 03/26/2024, 10:16:55 AM     LOG [LibraryService] Refreshing library: bbd31d37-7ef5-4927-8af1-f58efbe2b433
[Nest] 160  - 03/26/2024, 10:16:31 AM     LOG [CommunicationRepository] Websocket Connect:    clvldVUWtfcYvgRoAAAD
[Nest] 160  - 03/26/2024, 10:16:31 AM     LOG [CommunicationRepository] Websocket Disconnect: hnxZ0avJmzYXorD7AAAB
[Nest] 160  - 03/26/2024, 10:00:51 AM     LOG [CommunicationRepository] Websocket Connect:    hnxZ0avJmzYXorD7AAAB`

My last working scan was at the 16th of March
![grafik](https://github.com/immich-app/immich/assets/104008018/1fde1098-3c13-4e81-87c5-33cadf970b7c)

The external Library is setup as you can see in the picture:
![grafik](https://github.com/immich-app/immich/assets/104008018/62f26f9c-ec60-47ca-ae26-688c8daad1d8)

Here you can see the volumes of the Container:
`    volumes:
      - /sharedfolders/docker/settings/immich/config:/config
      - /sharedfolders/docker/settings/immich/photos:/photos
      - /sharedfolders/docker/settings/immich/machine-learning:/config/machine-learning #optional
      - /sharedfolders/docker/settings/immich/import:/import:ro #optional
      - /sharedfolders/bilder/Privat:/photos_library:ro   `

Does anyone have any ideas?

Big Thanks
V

**Comment by JW-CH** [comment]
Could you show you .env configuration? Did you set 'IMMICH_MEDIA_LOCATION'? As the message says, the media upload folder cannot be used for external libraries.

**Reply by kingvuke** [reply]
I do not have an .env file. Everything is here in my stack:

`# https://github.com/imagegenius/docker-immich
version: "2.1"
services:
  immich:
    image: ghcr.io/imagegenius/immich:latest
    #image: ghcr.io/imagegenius/immich:1.90.2
    container_name: immich_immich
    depends_on:
      - redis
      - postgres15
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=Europe/Berlin
      - DB_HOSTNAME=192.168.32.2
      - DB_USERNAME=postgres
      - DB_PASSWORD=XXXXXXXXXXXXXXXX
      - DB_DATABASE_NAME=immich
      - REDIS_HOSTNAME=192.168.32.3
      - DISABLE_MACHINE_LEARNING=false #optional
      - DISABLE_TYPESENSE=false #optional
      - DB_PORT=5432 #optional
      - REDIS_PORT=6379 #optional
      - REDIS_PASSWORD= #optional
      - MACHINE_LEARNING_WORKERS=1 #optional
      - MACHINE_LEARNING_WORKER_TIMEOUT=120 #optional
    volumes:
      - /sharedfolders/docker/settings/immich/config:/config
      - /sharedfolders/docker/settings/immich/photos:/photos
      - /sharedfolders/docker/settings/immich/machine-learning:/config/machine-learning #optional
      - /sharedfolders/docker/settings/immich/import:/import:ro #optional
      - /sharedfolders/bilder/Privat:/photos_library:ro      
    ports:
      - 2343:8080
    restart: unless-stopped
# This container requires an external application to be run separately to be run separately.
# Redis:
  redis:
    image: redis
    container_name: immich_redis
    depends_on:

      - postgres15
    ports:
      - 6379:6379
# PostgreSQL 15:
  postgres15:
    #image: postgres:15
    #image: tensorchord/pgvecto-rs:pg15-v0.1.11
    image: tensorchord/pgvecto-rs:pg15-v0.2.0
    container_name: immich_postgres15
    ports:
      - 5432:5432
    restart: unless-stopped
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: XXXXXXXXXXXXXXXX
      POSTGRES_DB: immich
    volumes:
      - /sharedfolders/docker/settings/immich/postgres:/var/lib/postgresql/data
      - `


And this are further settings in the Docker Container:
PUID=1000
TZ=Europe/Berlin
REDIS_PASSWORD=
REDIS_PORT=6379
REDIS_HOSTNAME=192.168.32.3
DB_PORT=5432
DISABLE_TYPESENSE=false
MACHINE_LEARNING_WORKERS=1
DB_DATABASE_NAME=immich
PGID=1000
DISABLE_MACHINE_LEARNING=false
DB_USERNAME=postgres
MACHINE_LEARNING_WORKER_TIMEOUT=120
DB_HOSTNAME=192.168.32.2
DB_PASSWORD=XXXXXXXXXXXXXXXXXXXX
PATH=/lsiopy/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
HOME=/root
LANGUAGE=en_US.UTF-8
LANG=en_US.UTF-8
TERM=xterm
S6_CMD_WAIT_FOR_SERVICES_MAXTIME=0
S6_VERBOSITY=1
S6_STAGE2_HOOK=/docker-mods
VIRTUAL_ENV=/lsiopy
IMMICH_MACHINE_LEARNING_URL=http://127.0.0.1:3003
IMMICH_MEDIA_LOCATION=/photos
IMMICH_WEB_ROOT=/app/immich/server/www
MACHINE_LEARNING_CACHE_FOLDER=/config/machine-learning/models
NVIDIA_DRIVER_CAPABILITIES=compute,video,utility
SERVER_PORT=8080
TRANSFORMERS_CACHE=/config/machine-learning/models
NODE_ENV=production



Hmmmm, chaning from image: ghcr.io/imagegenius/immich:latest to image: ghcr.io/imagegenius/immich:1.98.0 and it's working again:

![grafik](https://github.com/immich-app/immich/assets/104008018/09c6b730-4892-4254-9d1e-7da5e3c45863)

**Comment by JW-CH** [comment]
Ok, by looking at the code, I assume this is a bug.

The `isImmichPath()` method checks if your import path ('/photos_library') starts with the `IMMICH_MEDIA_LOCATION`-Variable ('/photos')
https://github.com/immich-app/immich/blob/9b705e445076db9206bc7ee4b0a878372819de60/server/src/cores/storage.core.ts#L117-L119

In my opinion, this is not done well, and the method should check, if the import path is a subfolder of `resolve(APP_MEDIA_LOCATION)` and not just starts with.

We might need a maintainer to confirm, but that's just how I see it.

**Reply by JW-CH** [reply]
See Issue #8319

**Reply by kingvuke** [reply]
Thank you very much for your support!
